rustls/src/msgs/deframer.rs

```diff
@@ -438,22 +439,26 @@ pub struct DeframerSliceBuffer<'a> {
     // a fully initialized buffer that will be deframed
     buf: &'a mut [u8],
     // number of bytes to discard from the front of `buf` at a later time
-    discard: usize,
```

Comment:
I think I'm not following you. The `discard` field type is changed to `Cell<usize>` on this commit basically. this is done to avoid borrowing issues later but I think adding that to a later PR would make it a little bit confusing. I'll add an explicit `groundwork:` to it.